benchmarks/babelstream/CMakeLists.txt

```
@@ -19,9 +19,30 @@ if(NOT TARGET alpaka::alpaka)
     endif()
 endif()
 
-alpaka_add_executable(${PROJECT_NAME} src/main.cpp src/Stream.h src/AlpakaStream.cpp src/AlpakaStream.h)
-target_compile_definitions(${PROJECT_NAME} PUBLIC ALPAKA)
-target_link_libraries(${PROJECT_NAME} PUBLIC alpaka::alpaka)
-set_target_properties(${PROJECT_NAME} PROPERTIES FOLDER benchmarks)
 
-# add_test(NAME ${_TARGET_NAME} COMMAND ${_TARGET_NAME})
+set(_TARGET_NAME "babelstream")
+append_recursive_files_add_to_src_group("src/" "src/" "cpp" _FILES_SOURCE)
+
+alpaka_add_executable(
+    ${_TARGET_NAME}
+    ${_FILES_SOURCE})
+
+target_include_directories(
+ ${_TARGET_NAME}
+ PRIVATE "src")
+
+target_link_libraries(
+    ${_TARGET_NAME}
+    PRIVATE common)
+
+#Run as a ctest
+if(alpaka_CI)
+    # Only run for release builds since this is a benchmark
+    if(CMAKE_BUILD_TYPE STREQUAL "Release")
+       set_target_properties(${_TARGET_NAME} PROPERTIES FOLDER benchmarks/babelstream)
+       add_test(NAME ${_TARGET_NAME} COMMAND ${_TARGET_NAME} --benchmark-samples 1 -r xml)
```

```suggestion
       add_test(NAME ${_TARGET_NAME} COMMAND ${_TARGET_NAME} --benchmark-samples 1 -r xml > /dev/null)
```

Otherwise the xml is displayed in the job output, which does not give use a benefit.

---

Even though it is not run exclusively it is still an indicator ?

---

That's true, but I expected that if something is going wrong with the CI, the babelstream executable returns a non 0 zero and die CI will fail. In this case we can remove temporary the `> /dev/null` to display the output.

Also I think it makes more sense, to create a custom job for this https://github.com/alpaka-group/alpaka/pull/2313 and checks the json output afterwards. Therefore the regular jobs should only check, if the benchmark is running with the XML reporter, but don't care about the output.

---

ok, i committed your suggestion.

---

tests started failing after adding >dev/null , local test also fails

---

I opened a PR (#2322), which suppresses ctest output in general, if the test didn't failed. Therefore you do not need to redirect the output to `/dev/null`.

---

I removed the dev/null